sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/UcfgExpressionService.cs

```diff
@@ -19,31 +19,66 @@
  */
 
 using System.Collections.Generic;
+using System.Linq;
 using Microsoft.CodeAnalysis;
 using SonarAnalyzer.Helpers;
-using SonarAnalyzer.Protobuf.Ucfg;
 
 namespace SonarAnalyzer.ControlFlowGraph.CSharp
 {
     internal class UcfgExpressionService
     {
-        private readonly Dictionary<SyntaxNode, Expression> syntaxNodeToUcfgExpressionCache
-            = new Dictionary<SyntaxNode, Expression>();
+        private static readonly ISet<KnownType> PrimitiveTypes = new[] { KnownType.System_Boolean }
```

Comment:
Minor question about naming: is the important thing that these are primitive types, or is that these are the variable types we handle in the CFG that just happen to all be primitive types at the moment? If is the latter then something like _SupportedVariableTypes_ might be a better name.